src/components/CardList.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, useState } from 'react';
+
+import classes from './CardList.module.css';
+import Card from './Card';
+import Draggable from './Draggable';
+import { CardListItemProps, CardListProps } from './CardList.types';
+import Modal from './Modal';
+import ItemDetails from './ItemDetails';
+import '../utils/array';
+
+const CardList: FC<CardListProps> = ({ cards, onUpdated }): React.ReactElement => {
+  const [dragginItemId, setDragginItemId] = useState('');
+  const [dragOver, setDragOver] = useState('');
+
+  const handleDragStart = (e: React.DragEvent<HTMLElement>): void => {
+    setDragginItemId(e.currentTarget.id);
+  };
+
+  const handleDragOver = (e: React.DragEvent<HTMLElement>): void => {
+    e.preventDefault();
+    setDragOver(e.currentTarget.id);
+  };
+
+  const handleDragEnd = (): void => {
+    setDragginItemId('');
+    setDragOver('');
+  };
+
+  const handleDrop = (e: React.DragEvent<HTMLElement>): void => {
+    const srcIndex = cards.findIndex((item) => item.id === dragginItemId);
```

Comment:
I'd suggest naming like `startIndex` and `endIndex` but it's NIT

Reply:
variable names were update